presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisSplitManager.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kinesis;
+
+import com.amazonaws.services.kinesis.model.DescribeStreamRequest;
+import com.amazonaws.services.kinesis.model.DescribeStreamResult;
+import com.amazonaws.services.kinesis.model.ResourceNotFoundException;
+import com.amazonaws.services.kinesis.model.Shard;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorSplit;
+import io.prestosql.spi.connector.ConnectorSplitManager;
+import io.prestosql.spi.connector.ConnectorSplitSource;
+import io.prestosql.spi.connector.ConnectorTableLayoutHandle;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.FixedSplitSource;
+
+import javax.inject.Named;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Split data chunk from kinesis Stream to multiple small chunks for parallelization and distribution to multiple Presto workers.
+ * By default, each shard of Kinesis Stream forms one Kinesis Split
+ */
+public class KinesisSplitManager
+        implements ConnectorSplitManager
+{
+    /**
+     * Max age of the shard cache (currently 24 hours).
+     */
+    public static final long MAX_CACHE_AGE_MILLIS = 24 * 3600 * 1000;
+
+    private final String connectorId;
+    private final KinesisHandleResolver handleResolver;
+    private final KinesisClientProvider clientManager;
+
+    private Map<String, InternalStreamDescription> streamMap = Collections.synchronizedMap(new HashMap<String, InternalStreamDescription>());
+
+    /**
+     * Cache the result of a Kinesis describe stream call so we don't need to retrieve
+     * the shards for every single query.
+     */
+    public static class InternalStreamDescription
+    {
+        private String streamName = "";
```

Comment:
These can all be final. Remove the initializers since they are set in the constructor.